dev/benchmarks/macrobenchmarks/lib/src/web/recorder.dart

```diff
@@ -225,10 +288,15 @@ abstract class WidgetRecorder extends Recorder implements _RecordingWidgetsBindi
 
   @override
   Future<Profile> run() {
+    profile = Profile(name: name);
     final _RecordingWidgetsBinding binding =
         _RecordingWidgetsBinding.ensureInitialized();
     final Widget widget = createWidget();
     binding._beginRecording(this, widget);
+
+    _profileCompleter.future.whenComplete(() {
+      profile = null;
+    });
```

Comment:
Since the profile is now a field on the recorder itself, can we make it a `Future<void>` and only use it as a signal? Otherwise we have multiple sources of the same object.

Reply:
What do you mean by "use it as a signal"?

I thought about that too, but decided not to do it because:
1. `profile` isn't a field in all recorders (see `RawRecorder`). It happens to be a field here because it is needed in various lifecycle method.
2. My understanding from the current design is that it's fair game to call `run` multiple times on the same recorder instance, in which case, it makes more sense to return `Future<Profile>` from `run()`.